veniq/dataset_mining/git_api.py

```diff
@@ -0,0 +1,271 @@
+                download_file(
+                    repo_name,
+                    commit_sha_before,
+                    filename_raw,
+                    output_dir_for_saved_file,
+                    headers,
+                    session,
+                    '_before',
+                    dict_result,
+                    True
+                )
+
+
+def handle_commit_example(sample, token, output_dir, classes_dict):
+    refactorings = sample.get('refactorings')
+    results = []
+    repository_url = sample['repository']
+    example_id = sample['id']
+    commit_sha_after = sample['sha1']
+    report_part_1 = Path(repository_url).parts[2:-1]
+    report_part_2 = Path(repository_url).parts[-1].replace('.git', '')
+    repo_name = Path(*report_part_1, report_part_2).as_posix()
+    description = refactorings.get('description')
+    filename_raw = Path(description.split('in class')[1].replace('.', '/').strip())
+    class_name = filename_raw.stem
+    classes = classes_dict.get(commit_sha_after, set())
+    if class_name not in classes:
+        add_to_dict_set(commit_sha_after, class_name, classes_dict)
+        output_dir_for_saved_file = output_dir / str(example_id)
+        if not output_dir_for_saved_file.exists():
+            output_dir_for_saved_file.mkdir(parents=True)
+
+        headers = {f"Authorization": f"token {token}"}
+        session = requests.Session()
+        d = {
+            'repo_url': repository_url,
+            'file_name_before': 'Not found',
+            'file_name_after': 'Not found',
+            'downloaded_after': False,
+            'downloaded_before': False,
+            'class_name': class_name,
+            'commit_before': '',
+            'commit_after': commit_sha_after,
+            'id': example_id,
+            'error': ''
+        }
+
+        download_file(
+            repo_name,
+            commit_sha_after,
+            filename_raw,
+            output_dir_for_saved_file,
+            headers,
+            session,
+            '_after',
+            d
+        )
+        results.append(d)
+
+    return results
+
+
+def search_filenames_in_commit(filename_raw: Path, files):
+    """
+    Searches filename in commit. If it is not found it tries
+    to find a subclass
+    :param filename_raw: file path of file in a commit
+    :param files: list of all items of a commit, given by github API
+    :return: list of found files
+    """
+    filename_to_search = Path(*filename_raw.parts[:-1], filename_raw.parts[-1] + '.java')
+    files_after_arr = find_filename(filename_to_search, files)
+    if not files_after_arr:
+        # finds subclass
+        filename_to_search = Path(*filename_raw.parts[:-2], filename_raw.parts[-2] + '.java')
+        files_after_arr = find_filename(filename_to_search, files)
+    if not files_after_arr:
+        # finds subclass of subclass
+        filename_to_search = Path(*filename_raw.parts[:-3], filename_raw.parts[-3] + '.java')
```

Comment:
The creation of subclass path looks quite weird. Can you make it more explicit, so it would be easier to understand why are you doing this? May be a comment can help?